src/js/media-trimmer.js

```diff
@@ -0,0 +1,493 @@
+      border: 2px solid #fff;
+      border-radius: 2px;
+      background-color: #000;
+      width: 160px;
+      height: 90px;
+
+      /* Negative offset of half to center on the handle */
+      margin-left: -80px;
+    }
+
+    /* Can't get this working. Trying a downward triangle. */
+    /* media-thumbnail-preview::after {
+      content: "";
+      display: block;
+      width: 300px;
+      height: 300px;
+      margin: 100px;
+      background-color: #ff0;
+    } */
+
+    :host(:hover) #thumbnailContainer.enabled {
+      display: block;
+      animation: fadeIn ease 0.5s;
+    }
+
+    @keyframes fadeIn {
+      0% {
+        /* transform-origin: bottom center; */
+        /* transform: scale(0.7); */
+        margin-top: 10px;
+        opacity: 0;
+      }
+      100% {
+        /* transform-origin: bottom center; */
+        /* transform: scale(1); */
+        margin-top: 0;
+        opacity: 1;
+      }
+    }
+  </style>
+  <div id="thumbnailContainer">
+    <media-thumbnail-preview></media-thumbnail-preview>
+  </div>
+  <div id="trimmerContainer">
+    <div id="timeline"></div>
+    <div id="playhead"></div>
+    <div id="leftTrim"></div>
+    <div id="selection">
+      <div id="startHandle"></div>
+      <div id="spacer"></div>
+      <div id="endHandle"></div>
+    </div>
+  </div>
+`;
+
+class MediaTrimmer extends MediaChromeHTMLElement {
+  static get observedAttributes() {
+    return ['thumbnails'].concat(super.observedAttributes || []);
+  }
+
+  constructor() {
+    super();
+
+    var shadow = this.attachShadow({ mode: 'open' });
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+
+    this.draggingEl = null;
+
+    this.wrapper = this.shadowRoot.querySelector('#trimmerContainer');
+    this.selection = this.shadowRoot.querySelector('#selection');
+    this.playhead = this.shadowRoot.querySelector('#playhead');
+    this.leftTrim = this.shadowRoot.querySelector('#leftTrim');
+    this.spacerFirst = this.shadowRoot.querySelector('#spacerFirst');
+    this.startHandle = this.shadowRoot.querySelector('#startHandle');
+    this.spacerMiddle = this.shadowRoot.querySelector('#spacerMiddle');
+    this.endHandle = this.shadowRoot.querySelector('#endHandle');
+    this.spacerLast = this.shadowRoot.querySelector('#spacerLast');
+
+    this._clickHandler = this.handleClick.bind(this);
+    this._dragStart = this.dragStart.bind(this);
+    this._dragEnd = this.dragEnd.bind(this);
+    this._drag = this.drag.bind(this);
+    /*
+     * TODO - teardown these handlers later
+     */
+    this.wrapper.addEventListener('click', this._clickHandler, false);
+
+    this.wrapper.addEventListener('touchstart', this._dragStart, false);
+    this.wrapper.addEventListener('touchend', this._dragEnd, false);
+    this.wrapper.addEventListener('touchmove', this._drag, false);
+
+    this.wrapper.addEventListener('mousedown', this._dragStart, false);
+    this.wrapper.addEventListener('mouseup', this._dragEnd, false);
```

Comment:
Probably need to listen to window mouseup and mousemove. If you mouseup off the component today it breaks stuff.